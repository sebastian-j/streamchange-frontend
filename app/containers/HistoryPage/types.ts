interface IHistoryItem {
  id: number;
  channelId: string;
  createdAt: string;
  displayName: string;
  imageUrl: string;
  message: string;
  prize: string;
}

type HistoryItem = IHistoryItem;

export { HistoryItem };
