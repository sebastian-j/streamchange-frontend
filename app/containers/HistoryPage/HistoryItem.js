import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import RelativeDate from '../../components/RelativeDate';

const Row = styled.tr`
  height: 50px;
`;

const Cell = styled.td`
  border-bottom: 1px solid #5e5e5e;
  border-spacing: 0;
  border-collapse: collapse;
  color: ${(props) => props.theme.staticTextColor};
  font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
  ${({ image }) =>
    image &&
    `
    max-width: 100px;
  `}
  ${({ textLeft }) =>
    textLeft &&
    `
    font-size: 1rem;
    padding: 14px 30px 14px 16px;
    text-align: left;
  `}
  ${({ text }) =>
    text &&
    `
    font-size: 0.875rem;
    padding: 14px 30px 14px 16px;
    text-align: right;
  `}
`;

const HistoryItem = (props) => (
  <Row>
    <Cell image>
      <a
        href={`https://www.youtube.com/channel/${props.channelId}`}
        target="_blank"
      >
        <img alt="Logo" src={props.imageUrl} height="45px" />
      </a>
    </Cell>
    <Cell textLeft>
      <span>{props.displayName}</span>
    </Cell>
    <Cell text>
      <span>{props.prize}</span>
    </Cell>
    <Cell text>
      <span>{props.message}</span>
    </Cell>
    <Cell text>
      <RelativeDate ISO8601Date={props.createdAt} />
    </Cell>
  </Row>
);

HistoryItem.propTypes = {
  channelId: PropTypes.string.isRequired,
  imageUrl: PropTypes.string,
  displayName: PropTypes.string,
  prize: PropTypes.string,
  message: PropTypes.string,
  createdAt: PropTypes.string,
};
export default HistoryItem;
