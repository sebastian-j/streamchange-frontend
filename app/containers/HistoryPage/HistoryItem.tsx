import React from 'react';
import PropTypes from 'prop-types';

import { Cell } from './components/Cell';
import { Row } from './components/Row';
import RelativeDate from '../../components/RelativeDate';

const HistoryItem = (props) => (
  <Row>
    <Cell className="image">
      <a
        href={`https://www.youtube.com/channel/${props.channelId}`}
        target="_blank"
      >
        <img alt="Logo" src={props.imageUrl} height="45px" />
      </a>
    </Cell>
    <Cell className="textLeft">
      <span>{props.displayName}</span>
    </Cell>
    <Cell className="text">
      <span>{props.prize}</span>
    </Cell>
    <Cell className="text">
      <span>{props.message}</span>
    </Cell>
    <Cell className="text">
      <RelativeDate ISO8601Date={props.createdAt} />
    </Cell>
  </Row>
);

HistoryItem.propTypes = {
  channelId: PropTypes.string.isRequired,
  imageUrl: PropTypes.string,
  displayName: PropTypes.string,
  prize: PropTypes.string,
  message: PropTypes.string,
  createdAt: PropTypes.string,
};
export default HistoryItem;
