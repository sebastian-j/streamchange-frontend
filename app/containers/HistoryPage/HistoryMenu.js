import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { FormattedMessage } from 'react-intl';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import IconButton from '@material-ui/core/IconButton';
import MenuItem from '@material-ui/core/MenuItem';
import MenuList from '@material-ui/core/Menu';

import messages from './messages';
import db from '../../components/YoutubeWorker/db';

const ThemedSvg = styled.svg`
  color: ${(props) => props.theme.staticTextColor};
`;

const HistoryMenu = (props) => {
  const [anchorEl, setAnchorEl] = useState(null);
  const [isOpenDialog, setIsOpenDialog] = useState(false);

  const openMenu = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const closeMenu = () => {
    setAnchorEl(null);
  };

  const openDialog = () => {
    setAnchorEl(null);
    setIsOpenDialog(true);
  };

  const closeDialog = () => {
    setIsOpenDialog(false);
  };

  const clearHistory = () => {
    db.history.clear().then(() => props.onClear());
    closeDialog();
  };

  return (
    <div>
      <IconButton edge="end" aria-label="Options" onClick={openMenu}>
        <ThemedSvg
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 24 24"
          width="18px"
          height="18px"
        >
          <path d="M0 0h24v24H0z" fill="none" />
          <path d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z" />
        </ThemedSvg>
      </IconButton>
      <MenuList
        id="simple-menu"
        anchorEl={anchorEl}
        open={!!anchorEl}
        onClose={closeMenu}
      >
        <MenuItem onClick={openDialog}>
          <FormattedMessage {...messages.clearHistoryMenuItem} />
        </MenuItem>
      </MenuList>
      <Dialog
        open={isOpenDialog}
        onClose={closeDialog}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">
          <FormattedMessage {...messages.clearHistoryMenuItem} />
        </DialogTitle>
        <DialogContent>
          <DialogContentText>
            <FormattedMessage {...messages.clearHistoryDialogContent} />
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={closeDialog} color="secondary">
            <FormattedMessage {...messages.clearHistoryCancelButton} />
          </Button>
          <Button onClick={clearHistory} color="primary">
            <FormattedMessage {...messages.clearHistoryConfirmButton} />
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
};

HistoryMenu.propTypes = {
  onClear: PropTypes.func,
};

export default HistoryMenu;
