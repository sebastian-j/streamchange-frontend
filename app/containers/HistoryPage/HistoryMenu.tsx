import React, { useState } from 'react';
import styled from 'styled-components';
import { FormattedMessage } from 'react-intl';
import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogContentText from '@mui/material/DialogContentText';
import DialogTitle from '@mui/material/DialogTitle';
import IconButton from '@mui/material/IconButton';
import MenuItem from '@mui/material/MenuItem';
import MenuList from '@mui/material/Menu';

import messages from './messages';
import db from '../../components/YoutubeWorker/db';

const ThemedSvg = styled.svg`
  color: ${(props) => props.theme.staticTextColor};
`;

interface Props {
  onClear: () => void;
}

const HistoryMenu = ({onClear}: Props) => {
  const [anchorEl, setAnchorEl] = useState<Element | null>(null);
  const [isOpenDialog, setIsOpenDialog] = useState<boolean>(false);

  const openMenu = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const closeMenu = () => {
    setAnchorEl(null);
  };

  const openDialog = () => {
    setAnchorEl(null);
    setIsOpenDialog(true);
  };

  const closeDialog = () => {
    setIsOpenDialog(false);
  };

  const clearHistory = () => {
    db.table('history').clear().then(() => onClear());
    closeDialog();
  };

  return (
    <div>
      <IconButton edge="end" aria-label="Options" onClick={openMenu}>
        <ThemedSvg
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 24 24"
          width="18px"
          height="18px"
        >
          <path d="M0 0h24v24H0z" fill="none" />
          <path d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z" />
        </ThemedSvg>
      </IconButton>
      <MenuList
        id="simple-menu"
        anchorEl={anchorEl}
        open={!!anchorEl}
        onClose={closeMenu}
      >
        <MenuItem onClick={openDialog}>
          <FormattedMessage {...messages.clearHistoryMenuItem} />
        </MenuItem>
      </MenuList>
      <Dialog
        open={isOpenDialog}
        onClose={closeDialog}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">
          <FormattedMessage {...messages.clearHistoryMenuItem} />
        </DialogTitle>
        <DialogContent>
          <DialogContentText>
            <FormattedMessage {...messages.clearHistoryDialogContent} />
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={closeDialog} color="secondary">
            <FormattedMessage {...messages.clearHistoryCancelButton} />
          </Button>
          <Button onClick={clearHistory} color="primary">
            <FormattedMessage {...messages.clearHistoryConfirmButton} />
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
};

export default HistoryMenu;
