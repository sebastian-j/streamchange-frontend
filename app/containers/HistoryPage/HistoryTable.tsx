import React, { useState } from 'react';
import clsx from 'clsx';
import { FormattedMessage } from 'react-intl';

import messages from './messages';
import ArrowUpIcon from './components/arrowUpIcon';
import HistoryItem from './HistoryItem';
import { Table } from './components/Table';
import { TableHeaderButton } from './components/TableHeaderButton';
import { Thead } from './components/Thead';
import { HistoryItem as HItem } from './types';

interface Props {
  items: HItem[];
}

const HistoryTable = ({ items }: Props) => {
  const [sort, setSort] = useState<string>('createdAtDESC');
  const handleSortChange = (event) => {
    const { target } = event;
    let value = target.id;
    if (value === sort) {
      value += 'DESC';
    }
    setSort(value);
  };

  return (
    <Table>
      <Thead>
        <tr>
          <td />
          <td>
            <TableHeaderButton
              id="displayName"
              onClick={handleSortChange}
              type="button"
            >
              <FormattedMessage {...messages.nameHeader} />
              <ArrowUpIcon
                className={clsx(
                  sort === 'displayName' && 'active',
                  sort === 'displayNameDESC' && ['active', 'descending'],
                )}
              />
            </TableHeaderButton>
          </td>
          <td>
            <TableHeaderButton
              id="prize"
              onClick={handleSortChange}
              type="button"
            >
              <FormattedMessage {...messages.prizeHeader} />
              <ArrowUpIcon
                className={clsx(
                  sort === 'prize' && 'active',
                  sort === 'prizeDESC' && ['active', 'descending'],
                )}
              />
            </TableHeaderButton>
          </td>
          <td>
            <TableHeaderButton
              id="message"
              onClick={handleSortChange}
              type="button"
            >
              <FormattedMessage {...messages.messageHeader} />
              <ArrowUpIcon
                className={clsx(
                  sort === 'message' && 'active',
                  sort === 'messageDESC' && ['active', 'descending'],
                )}
              />
            </TableHeaderButton>
          </td>
          <td>
            <TableHeaderButton
              id="createdAt"
              onClick={handleSortChange}
              type="button"
            >
              <FormattedMessage {...messages.dateHeader} />
              <ArrowUpIcon
                className={clsx(
                  sort === 'createdAt' && 'active',
                  sort === 'createdAtDESC' && ['active', 'descending'],
                )}
              />
            </TableHeaderButton>
          </td>
        </tr>
      </Thead>
      <tbody>
        {items.map((item) => (
          <HistoryItem
            key={item.createdAt}
            channelId={item.channelId}
            imageUrl={item.imageUrl}
            displayName={item.displayName}
            prize={item.prize}
            message={item.message}
            createdAt={item.createdAt}
          />
        ))}
      </tbody>
    </Table>
  );
};

export default HistoryTable;
