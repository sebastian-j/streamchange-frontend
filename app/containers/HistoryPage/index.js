import React from 'react';
import { NavLink } from 'react-router-dom';
import styled from 'styled-components';
import LinearProgress from '@material-ui/core/LinearProgress';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import IconButton from '@material-ui/core/IconButton';
import { FormattedMessage } from 'react-intl';

import messages from './messages';
import db from '../../components/YoutubeWorker/db';
import StyledTextField from '../../components/StyledTextField';
import StyledFormControl from '../../components/StyledTextField/StyledFormControl';
import HistoryMenu from './HistoryMenu';
import HistoryTable from './HistoryTable';

const PageWrapper = styled.div`
  background-color: ${(props) => props.theme.bodyBackground};
  padding: 10px;
  overflow-x: hidden;
  min-height: 100vh;
  width: 100%;
`;

const Header = styled.header`
  display: flex;
  flex-direction: row;
`;

const ReturnButton = styled(NavLink)`
  display: block;
  color: ${(props) => props.theme.staticTextColor};
  flex-grow: 1;
  line-height: 48px;
  min-height: 48px;
  text-decoration: none;
  & span {
    margin-left: 24px;
  }
  &:hover {
    background: rgba(0, 0, 0, 0.2);
  }
`;

const Information = styled.div`
  color: ${(props) => props.theme.staticTextColor};
  font-size: 2vw;
  margin-top: 5vh;
  text-align: center;
`;
const Footer = styled.div`
  display: flex;
  flex-direction: row-reverse;
  margin-top: 10px;
  max-width: 200px;
`;

export default class HistoryPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      error: false,
      isLoaded: false,
      items: [],
      search: '',
      maxResults: 20,
      page: 0,
      isLastPage: false,
    };
    this.handleChange = this.handleChange.bind(this);
    this.prevPage = this.prevPage.bind(this);
    this.nextPage = this.nextPage.bind(this);
    this.getHistory = this.getHistory.bind(this);
  }

  handleChange(event) {
    const { target } = event;
    const { value } = target;
    const { name } = target;
    this.setState(
      {
        [name]: value,
      },
      () => {
        this.getHistory();
      },
    );
  }

  prevPage() {
    const oldPage = this.state.page;
    if (this.state.page > 0) {
      this.setState({ page: oldPage - 1 }, () => {
        this.getHistory();
      });
    }
  }

  nextPage() {
    const oldPage = this.state.page;
    this.setState({ page: Number(oldPage + 1) }, () => {
      this.getHistory();
    });
  }

  getHistory() {
    const firstResult = Number(this.state.page * this.state.maxResults);
    db.table('history')
      .filter((winner) =>
        winner.displayName
          .toLowerCase()
          .includes(this.state.search.toLowerCase()),
      )
      .reverse()
      .offset(firstResult)
      .limit(this.state.maxResults)
      .toArray()
      .then((items) => {
        this.setState({ isLoaded: true, error: false, items });
        if (this.state.maxResults > items.length) {
          this.setState({ isLastPage: true });
        } else this.setState({ isLastPage: false });
      })
      .catch(() => {
        this.setState({
          isLoaded: true,
          error: true,
        });
      });
  }

  componentDidMount() {
    this.getHistory();
  }

  render() {
    if (this.state.error) {
      return (
        <div>
          <FormattedMessage {...messages.infoError} />
        </div>
      );
    }
    if (!this.state.isLoaded) {
      return (
        <PageWrapper>
          <LinearProgress />
          <div
            style={{
              fontSize: '2vw',
              position: 'absolute',
              left: '55%',
              top: '50%',
              transform: 'translate(-50%, -50%)',
              textAlign: 'center',
            }}
          >
            <FormattedMessage {...messages.infoLoading} />
          </div>
        </PageWrapper>
      );
    }
    return (
      <PageWrapper>
        <Header>
          <ReturnButton to="/giveaway" activeClassName="active">
            <div>
              <svg
                xmlns="http://www.w3.org/2000/svg"
                width="24"
                height="24"
                viewBox="0 0 24 24"
              >
                <path d="M0 0h24v24H0z" fill="none" />
                <path d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z" />
              </svg>
              <span>
                <FormattedMessage {...messages.returnButton} />
              </span>
            </div>
          </ReturnButton>
          <HistoryMenu onClear={this.getHistory} />
        </Header>
        <FormattedMessage {...messages.searchLabel}>
          {(label) => (
            <StyledTextField
              id="search"
              name="search"
              label={label}
              value={this.state.search}
              onChange={this.handleChange}
              type="text"
              margin="normal"
              fullWidth
            />
          )}
        </FormattedMessage>
        {this.state.search.length > 0 && this.state.items.length === 0 && (
          <Information>
            <FormattedMessage {...messages.infoNoResults} />
          </Information>
        )}
        {this.state.items.length > 0 && (
          <HistoryTable items={this.state.items} />
        )}
        <Footer>
          {!this.state.isLastPage && (
            <IconButton
              edge="end"
              aria-label="Next page"
              name="next"
              onClick={this.nextPage}
            >
              <svg
                xmlns="http://www.w3.org/2000/svg"
                width="24"
                height="24"
                viewBox="0 0 24 24"
              >
                <path d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z" />
                <path fill="none" d="M0 0h24v24H0V0z" />
              </svg>
            </IconButton>
          )}
          {this.state.page !== 0 && (
            <IconButton
              edge="end"
              aria-label="Previous page"
              name="prev"
              onClick={this.prevPage}
            >
              <svg
                xmlns="http://www.w3.org/2000/svg"
                width="24"
                height="24"
                viewBox="0 0 24 24"
              >
                <path d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z" />
                <path fill="none" d="M0 0h24v24H0V0z" />
              </svg>
            </IconButton>
          )}
          <StyledFormControl margin="normal">
            <InputLabel htmlFor="maxResults">
              <FormattedMessage {...messages.resultsPerPage} />
            </InputLabel>
            <Select
              value={this.state.maxResults}
              onChange={this.handleChange}
              inputProps={{
                name: 'maxResults',
                id: 'maxResults',
              }}
            >
              <MenuItem value={10}>10</MenuItem>
              <MenuItem value={20}>20</MenuItem>
              <MenuItem value={50}>50</MenuItem>
              <MenuItem value={100}>100</MenuItem>
            </Select>
          </StyledFormControl>
        </Footer>
      </PageWrapper>
    );
  }
}
