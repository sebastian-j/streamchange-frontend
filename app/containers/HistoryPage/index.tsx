import React, { useEffect, useState } from 'react';
import LinearProgress from '@material-ui/core/LinearProgress';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import IconButton from '@material-ui/core/IconButton';
import { FormattedMessage } from 'react-intl';

import messages from './messages';
import db from '../../components/YoutubeWorker/db';
import StyledTextField from '../../components/StyledTextField';
import StyledFormControl from '../../components/StyledTextField/StyledFormControl';
import HistoryMenu from './HistoryMenu';
import HistoryTable from './HistoryTable';
import { InformationText } from './components/InformationText';
import { PageHeader } from './components/PageHeader';
import { PageWrapper } from './components/PageWrapper';
import { ReturnButton } from './components/ReturnButton';
import { TableFooter } from './components/TableFooter';
import { HistoryItem } from './types';

const HistoryPage = () => {
  const [error, setError] = useState<boolean>(false);
  const [isLoaded, setIsLoaded] = useState<boolean>(false);
  const [items, setItems] = useState<HistoryItem[]>([]);
  const [searchQuery, setSearchQuery] = useState<string>('');
  const [maxResults, setMaxResults] = useState<number>(20);
  const [page, setPage] = useState<number>(0);
  const [isLastPage, setIsLastPage] = useState<boolean>(false);

  const getHistory = () => {
    const firstResult = Number(page * maxResults);
    db.table('history')
      .filter((winner) =>
        winner.displayName.toLowerCase().includes(searchQuery.toLowerCase()),
      )
      .reverse()
      .offset(firstResult)
      .limit(maxResults)
      .toArray()
      .then((it: HistoryItem[]) => {
        setIsLoaded(true);
        setError(false);
        setItems(it);
        if (maxResults > it.length) {
          setIsLastPage(true);
        } else setIsLastPage(false);
      })
      .catch(() => {
        setIsLoaded(true);
        setError(true);
      });
  };

  const prevPage = () => {
    if (page > 0) {
      setPage((prevState) => prevState - 1);
    }
  };

  const nextPage = () => {
    setPage((prevState) => prevState + 1);
  };

  useEffect(() => {
    getHistory();
  }, [maxResults]);
  useEffect(() => {
    getHistory();
  }, [page]);
  useEffect(() => {
    getHistory();
  }, [searchQuery]);
  useEffect(() => {
    getHistory();
  }, []);

  if (error) {
    return (
      <div>
        <FormattedMessage {...messages.infoError} />
      </div>
    );
  }
  if (!isLoaded) {
    return (
      <PageWrapper>
        <LinearProgress />
        <div
          style={{
            fontSize: '2vw',
            position: 'absolute',
            left: '55%',
            top: '50%',
            transform: 'translate(-50%, -50%)',
            textAlign: 'center',
          }}
        >
          <FormattedMessage {...messages.infoLoading} />
        </div>
      </PageWrapper>
    );
  }
  return (
    <PageWrapper>
      <PageHeader>
        <ReturnButton to="/giveaway" activeClassName="active">
          <div>
            <svg
              xmlns="http://www.w3.org/2000/svg"
              width="24"
              height="24"
              viewBox="0 0 24 24"
            >
              <path d="M0 0h24v24H0z" fill="none" />
              <path d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z" />
            </svg>
            <span>
              <FormattedMessage {...messages.returnButton} />
            </span>
          </div>
        </ReturnButton>
        <HistoryMenu onClear={getHistory} />
      </PageHeader>
      <FormattedMessage {...messages.searchLabel}>
        {(label) => (
          <StyledTextField
            id="search"
            name="search"
            label={label}
            value={searchQuery}
            onChange={(event) => setSearchQuery(event.target.value)}
            type="text"
            margin="normal"
            fullWidth
          />
        )}
      </FormattedMessage>
      {searchQuery.length > 0 && items.length === 0 && (
        <InformationText>
          <FormattedMessage {...messages.infoNoResults} />
        </InformationText>
      )}
      {items.length > 0 && <HistoryTable items={items} />}
      <TableFooter>
        {!isLastPage && (
          <IconButton
            edge="end"
            aria-label="Next page"
            name="next"
            onClick={nextPage}
          >
            <svg
              xmlns="http://www.w3.org/2000/svg"
              width="24"
              height="24"
              viewBox="0 0 24 24"
            >
              <path d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z" />
              <path fill="none" d="M0 0h24v24H0V0z" />
            </svg>
          </IconButton>
        )}
        {page !== 0 && (
          <IconButton
            edge="end"
            aria-label="Previous page"
            name="prev"
            onClick={prevPage}
          >
            <svg
              xmlns="http://www.w3.org/2000/svg"
              width="24"
              height="24"
              viewBox="0 0 24 24"
            >
              <path d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z" />
              <path fill="none" d="M0 0h24v24H0V0z" />
            </svg>
          </IconButton>
        )}
        <StyledFormControl margin="normal">
          <InputLabel htmlFor="maxResults">
            <FormattedMessage {...messages.resultsPerPage} />
          </InputLabel>
          <Select
            value={maxResults}
            onChange={(event) => {
              setMaxResults(Number(event.target.value));
            }}
            inputProps={{
              name: 'maxResults',
              id: 'maxResults',
            }}
          >
            <MenuItem value={10}>10</MenuItem>
            <MenuItem value={20}>20</MenuItem>
            <MenuItem value={50}>50</MenuItem>
            <MenuItem value={100}>100</MenuItem>
          </Select>
        </StyledFormControl>
      </TableFooter>
    </PageWrapper>
  );
};

export default HistoryPage;
