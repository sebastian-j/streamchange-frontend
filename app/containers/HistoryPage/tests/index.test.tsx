import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import HistoryPage from '../index';

const shallowRenderer = createRenderer();

describe('<HistoryPage />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(<HistoryPage />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
