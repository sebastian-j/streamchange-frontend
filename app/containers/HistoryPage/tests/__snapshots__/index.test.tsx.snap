// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HistoryPage /> should render and match the snapshot 1`] = `
<styled.div>
  <WithStyles(ForwardRef(LinearProgress)) />
  <div
    style={
      Object {
        "fontSize": "2vw",
        "left": "55%",
        "position": "absolute",
        "textAlign": "center",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
      }
    }
  >
    <FormattedMessage
      defaultMessage="Loading history"
      id="streamchange.containers.History.info.loading"
      values={Object {}}
    />
  </div>
</styled.div>
`;
